internal/services/cio/cio.go

```diff
@@ -0,0 +1,83 @@
+package cio
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pb_accounts "github.com/DIMO-Network/accounts-api/pkg/grpc"
+	"github.com/DIMO-Network/devices-api/models"
+	analytics "github.com/customerio/cdp-analytics-go"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+)
+
+const SoftwareConnectionExpiredEvent = "software_connection_expired"
+
+type AccountsClient interface {
+	GetAccount(context.Context, *pb_accounts.GetAccountRequest, ...grpc.CallOption) (*pb_accounts.Account, error)
+}
+
+type Service struct {
+	client     analytics.Client
+	logger     *zerolog.Logger
+	acctClient AccountsClient
+}
+
+func New(cioKey string, acctClient AccountsClient, logger *zerolog.Logger) (*Service, error) {
+	client, err := analytics.NewWithConfig(cioKey, analytics.Config{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		client:     client,
+		acctClient: acctClient,
+		logger:     logger,
+	}, nil
+
+}
+
+func (s *Service) SoftwareDisconnectionEvent(ctx context.Context, udai *models.UserDeviceAPIIntegration) error {
+	vehicleTokenID, ok := udai.R.UserDevice.TokenID.Int64()
```

Comment:
Let's make sure that this thing has a token id in the first place. There are probably still some vestigial rows.